lib/state_change_syncer.go

```diff
@@ -672,10 +674,10 @@ func (stateChangeSyncer *StateChangeSyncer) SyncMempoolToStateSyncer(server *Ser
 	}
 
 	// Loop through all the transactions in the mempool and connect them and their utxo ops to the mempool view.
-	server.mempool.mtx.RLock()
-	mempoolTxns, _, err := server.mempool._getTransactionsOrderedByTimeAdded()
-	server.mempool.mtx.RUnlock()
+	mempoolTxns := server.GetMempool().GetOrderedTransactions()
```

Comment:
Consider capping the number of txns returned by GetOrderedTransactions. If the mempool is flooded with many transactions, then the SyncMempoolToStateSyncer function can choke